### PR TITLE
Migrate 'taxonomies' e2e tests to Playwright

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #38851.

PR migrates `taxonomies.test.js` e2e tests to Playwright.

## Why?
See #38851.

## Notes
* I dropped `if the user has the right capabilities` suffixes and `canCreatTermInTaxonomy` checks. The tests currently only run for admin users which have the capabilities.

## Testing Instructions
```
npm run test:e2e:playwright -- test/e2e/specs/editor/various/taxonomies.spec.js
```

### CODE REVIEW

File: test/e2e/specs/editor/various/taxonomies.spec.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		} );
+
+		if (
+			( await panelToggle.getAttribute( 'aria-expanded' ) ) === 'false'
+		) {
+			await panelToggle.click();
+		}
+
+		const tagName = 'tag-' + generateRandomNumber();
+		const tags = page.locator( '.components-form-token-field__token-text' );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/taxonomies.spec.js:80]: I couldn't find a better locator here. The screen readers use the `Remove Tag` button and its description, but PW currently can't locate elements based on descriptions.

Some workarounds are described in https://github.com/microsoft/playwright/issues/18332, but decided to go with the simplest solution.

File: test/e2e/specs/editor/various/taxonomies.spec.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+		if (
+			( await panelToggle.getAttribute( 'aria-expanded' ) ) === 'false'
+		) {
+			await panelToggle.click();
+		}
+
+		await page
+			.getByRole( 'button', {
+				name: 'Add New Category',
+				expanded: false,
```
Comment (t-hamano) [inline test/e2e/specs/editor/various/taxonomies.spec.js:34]: This line seems unnecessary.

**Review by t-hamano — APPROVED** [review]
I left only one comment, but LGTM 👍

File: test/e2e/specs/editor/various/taxonomies.spec.js (hunk 2 above)
Comment (Mamaduka) [reply test/e2e/specs/editor/various/taxonomies.spec.js:34]: The panel has two `Add New Category` buttons; the first expands the form, and the second submits. I added the expanded check as a safeguard and to highlight the intention.

## Screenshot
CleanShot 2024-01-09 at 12 08 23

Comment (t-hamano) [reply test/e2e/specs/editor/various/taxonomies.spec.js:34]: Oh, I missed that 😅 If so, there is no problem.

**Pull request merged**